language/tools/test-generation/src/main.rs

```diff
@@ -8,21 +8,36 @@ use std::env;
 use structopt::StructOpt;
 use test_generation::{config::Args, run_generation};
 
-fn setup_log() {
+fn setup_log(log_file: &Option<String>) {
     if env::var("RUST_LOG").is_err() {
         env::set_var("RUST_LOG", "info");
     }
-    let decorator = slog_term::PlainDecorator::new(std::io::stdout());
-    let drain = slog_term::CompactFormat::new(decorator).build().fuse();
-    let drain = slog_envlogger::new(drain);
-    let drain = std::sync::Mutex::new(drain).fuse();
-    let logger = slog::Logger::root(drain, o!());
-    let logger_guard = slog_scope::set_global_logger(logger);
-    std::mem::forget(logger_guard);
+    if let Some(log_file_path) = log_file {
+        let file = std::fs::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(log_file_path)
+            .unwrap();
+        let flog = slog_json::Json::default(file).fuse();
+        let drain = slog_envlogger::new(flog);
+        let drain = std::sync::Mutex::new(drain).fuse();
```

Comment:
Many lines in common between then and else branch.  Some refactoring would help.